### PR TITLE
Reworked get department endpoint

### PR DESCRIPTION
**Pull request by djino97**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/784

### CODE REVIEW

File: src/CompanyService/Startup.cs [hunk 1]
```diff
@@ -111,7 +111,7 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
 
             app.UseRouting();
 
-            app.UseMiddleware<TokenMiddleware>();
+            //app.UseMiddleware<TokenMiddleware>();
```
Comment (MariyaSin) [inline src/CompanyService/Startup.cs:114]: remove //

**Review by MariyaSin — APPROVED** [review]

File: src/CompanyService/Startup.cs (hunk 1 above)
Comment (djino97) [reply src/CompanyService/Startup.cs]: Removed

File: src/CompanyService.Business/Commands/Department/Interfaces/IGetDepartmentCommand.cs [hunk 2]
```diff
@@ -9,14 +11,14 @@ namespace LT.DigitalOffice.CompanyService.Business.Commands.Department.Interface
...
+    public interface IGetDepartmentCommand
     {
         /// <summary>
         /// Returns the department model with the specified id.
         /// </summary>
         /// <param name="departmentId">Specified id of department.</param>
         /// <returns></returns>
         /// <exception cref="Kernel.Exceptions.NotFoundException">Thrown when departmen is not found.</exception>
-        DepartmentInfo Execute(Guid departmentId);
+        OperationResultResponse<DepartmentResponse> Execute(GetDepartmentFilter filter);
```
Comment (BelehovEgor) [inline src/CompanyService.Business/Commands/Department/Interfaces/IGetDepartmentCommand.cs:22]: if you change the method signature change the documentation too

File: src/CompanyService.Models.Dto/Models/DepartmentUserInfo.cs [hunk 3]
```diff
@@ -1,10 +1,10 @@
-﻿using System;
-
-namespace LT.DigitalOffice.CompanyService.Models.Dto.Models
+﻿namespace LT.DigitalOffice.CompanyService.Models.Dto.Models
 {
-    public record DepartmentUserInfo
+    public record DepartmentUserInfo : UserInfo
```
Comment (BelehovEgor) [inline src/CompanyService.Models.Dto/Models/DepartmentUserInfo.cs:3]: do you really want to do this?)

Comment (BelehovEgor) [reply src/CompanyService.Models.Dto/Models/DepartmentUserInfo.cs:3]: I am about inheritance

File: src/CompanyService.Models.Dto/Requests/Filters/GetDepartmentFilter.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace LT.DigitalOffice.CompanyService.Models.Dto.Requests.Filters
+{
+    public record GetDepartmentFilter
+    {
+        [FromQuery(Name ="departmentid")]
+        public Guid DepartmentId { get; set; }
+        public bool? IncludeUsers { get; set; }
```
Comment (BelehovEgor) [inline src/CompanyService.Models.Dto/Requests/Filters/GetDepartmentFilter.cs:11]: you forget about from query

File: src/CompanyService.Models.Dto/Responses/DepartmentResponse.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.CompanyService.Models.Dto.Models;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.CompanyService.Models.Dto.Responses
+{
+    public record DepartmentResponse
+    {
+        public ShortDepartmentInfo Department { get; set; }
+        public IEnumerable<DepartmentUserInfo> Users { get; set; }
```
Comment (BelehovEgor) [inline src/CompanyService.Models.Dto/Responses/DepartmentResponse.cs:9]: maybe we will use list?

**Review by BelehovEgor — APPROVED** [review]

File: src/CompanyService.Models.Dto/Models/DepartmentUserInfo.cs (hunk 3 above)
Comment (djino97) [reply src/CompanyService.Models.Dto/Models/DepartmentUserInfo.cs:3]: it is working now, if need be then we will rewrite in future

File: src/CompanyService.Models.Dto/Requests/Filters/GetDepartmentFilter.cs (hunk 4 above)
Comment (TIove) [inline src/CompanyService.Models.Dto/Requests/Filters/GetDepartmentFilter.cs:11]: includeusers

File: src/CompanyService.Business/Commands/Department/Interfaces/IGetDepartmentCommand.cs (hunk 2 above)
Comment (TIove) [reply src/CompanyService.Business/Commands/Department/Interfaces/IGetDepartmentCommand.cs:22]: Or delete it at all)

File: src/CompanyService.Models.Dto/Responses/DepartmentResponse.cs (hunk 5 above)
Comment (djino97) [reply src/CompanyService.Models.Dto/Responses/DepartmentResponse.cs:9]: not need

File: src/CompanyService.Models.Dto/Requests/Filters/GetDepartmentFilter.cs (hunk 4 above)
Comment (djino97) [reply src/CompanyService.Models.Dto/Requests/Filters/GetDepartmentFilter.cs:13]: Fixed

File: src/CompanyService.Business/Commands/Department/Interfaces/IGetDepartmentCommand.cs (hunk 2 above)
Comment (djino97) [reply src/CompanyService.Business/Commands/Department/Interfaces/IGetDepartmentCommand.cs:12]: Removed docs

**Review by TIove — APPROVED** [review]

**Review by TIove — APPROVED** [review]

**Pull request merged**